### ISSUE TITLE
[Sparse] Sparse Softmax CUDA Kernel

### ISSUE THREAD

**Issue by mufeili** [dmlc/dgl#5063]
## 🔨Work Item

Project tracker: https://github.com/orgs/dmlc/projects/2

## Description

#4866 will be closed by following the existing practice of `edge_softmax`, which invokes `gspmm`, `gsddmm`, `gspmm`, and `gsddmm` in order. In the long run, we should provide a single CUDA kernel.

References
- https://github.com/dmlc/dgl/pull/3826
- https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/utils/softmax.py
- https://github.com/pyg-team/pyg-lib/blob/f4ad48796a7683e09a8398daa8989273e0109c03/pyg_lib/ops/__init__.py#L132

**Comment by frozenbugs** [comment]
We don't foresee clear performance improvement opportunity here. The existing code https://github.com/dmlc/dgl/blob/master/dgl_sparse/src/softmax.cc style is pretty good, close this for now.
